app/src/main/java/org/pragyan/dalal18/fragment/OTPVerificationDialogFragment.kt

```diff
@@ -0,0 +1,131 @@
+import org.pragyan.dalal18.ui.MainActivity
+import org.pragyan.dalal18.ui.OtpEditText
+import org.pragyan.dalal18.ui.VerifyPhoneActivity
+import org.pragyan.dalal18.utils.ConnectionUtils
+import org.pragyan.dalal18.utils.Constants
+import org.pragyan.dalal18.utils.MiscellaneousUtils
+import org.pragyan.dalal18.utils.hideKeyboard
+import java.util.ArrayList
+import javax.inject.Inject
+
+class OTPVerificationDialogFragment : DialogFragment() {
+
+    @Inject
+    lateinit var channel: ManagedChannel
+
+    lateinit var mobNumber: String
+    lateinit var phoneNumberEditText: EditText
+    lateinit var resendOtp: Button
+    lateinit var verifyOtp: Button
+    lateinit var otpEditText: OtpEditText
+
+    companion object {
+        fun newInstance(phNum: String) : OTPVerificationDialogFragment  {
+
+            val f = OTPVerificationDialogFragment()
+            f.mobNumber = phNum
+
+            return f
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        DaggerDalalStreetApplicationComponent.builder().contextModule(ContextModule(context!!)).build().inject(this)
+
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val v = inflater.inflate(R.layout.layout_otp_verification_dialog, container, false)
+        phoneNumberEditText = v.findViewById(R.id.enter_otp_mobno_edit_text)
+        resendOtp = v.findViewById(R.id.btnResendOtp)
+        verifyOtp = v.findViewById(R.id.btnVerifyOtp)
+        otpEditText = v.findViewById(R.id.et_otp)
+        return v
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        phoneNumberEditText.setText(mobNumber)
+
+        resendOtp.setOnClickListener { sendOtpAgain() }
+        verifyOtp.setOnClickListener {
+            if (otpEditText.text!!.toString() == "")
+                Toast.makeText(context, "Enter OTP.", Toast.LENGTH_SHORT).show()
+            else
+                checkIfOtpIsCorrect(otpEditText.text!!.toString())
+        }
+    }
+
+    private fun checkIfOtpIsCorrect(OTP: String) = lifecycleScope.launch {
+
+        if (withContext(Dispatchers.IO) { ConnectionUtils.getConnectionInfo(context) }) {
+            val verifyOTPRequest = VerifyOTPRequest
+                    .newBuilder()
+                    .setOtp(Integer.parseInt(OTP))
+                    .build()
+
+            val verifyOTPResponse = withContext(Dispatchers.IO) { DalalActionServiceGrpc.newBlockingStub(channel).verifyOTP(verifyOTPRequest) }
+
+            Toast.makeText(context,verifyOTPResponse.statusMessage,Toast.LENGTH_SHORT).show()
+
+            if (verifyOTPResponse.statusCode == VerifyOTPResponse.StatusCode.OK) {
+                // go to main with all intent values
+                val intent = Intent(activity,MainActivity::class.java)
```

Comment:
Shouldn't we pass some extras through this intent, check Login Activity for all the extras passed. For example,

intent.putExtra(Constants.USERNAME_KEY, loginResponse.user.name)
intent.putExtra(MainActivity.CASH_WORTH_KEY, loginResponse.user.cash)
intent.putExtra(MainActivity.TOTAL_WORTH_KEY, loginResponse.user.total)
.
.
.